pkg/lib/authn/identity/oauth/provider.go

```diff
@@ -90,8 +91,15 @@ func (p *Provider) WithUpdate(
 	claims map[string]interface{},
 ) *identity.OAuth {
 	newIden := *iden
-	newIden.UserProfile = rawProfile
-	newIden.Claims = claims
+	// For non-Apple provider, we can just update.
+	// For Apple, we need to merge given_name and family_name because
+	// they only available at THE FIRST TIME authorization.
+	if newIden.ProviderID.Type == liboauthrelyingparty.TypeApple {
+		newIden.Apple_MergeRawProfileAndClaims(rawProfile, claims)
+	} else {
+		newIden.UserProfile = rawProfile
+		newIden.Claims = claims
+	}
```

Comment:
Or instead of having a special code path for apple, we can always "merge" the original claims and the new claims, by setting all existing keys in the new claims in the original claims, and ignore non-existing keys?

Reply:
> setting all existing keys in the new claims in the original claims, and ignore non-existing keys?

I didn't do this because

1. existing claims has key `a`
2. incoming claims DOES NOT have key `a`

Then the resulting claims should not have `a`.

But in case of Apple, we want the reverse, we want `given_name` and `family_name` to be present.